### PR TITLE
Dev 0.5

### PR DESCRIPTION
**Pull request by lindnemi**

### CODE REVIEW

File: src/NetworkStructures.jl [hunk 1]
```diff
@@ -102,8 +98,25 @@ function GraphStruct(g, v_dims, e_dims, v_syms, e_syms)
...
 
-    e_s_v_dat = [[(offset, dim) for (i_e, (offset, dim)) in enumerate(zip(e_offs, e_dims)) if i_v == s_e[i_e]] for i_v in 1:num_v]
-    e_d_v_dat = [[(offset, dim) for (i_e, (offset, dim)) in enumerate(zip(e_offs, e_dims)) if i_v == d_e[i_e]] for i_v in 1:num_v]
+    in_edges_dat = Vector{Vector{Tuple{Int,Int}}}(undef, nv(g))
+
+    for i_v in 1:nv(g)
+        offsdim_arr = Tuple{Int,Int}[]
+        for i_e in d_v[i_v]
+            # assert that dims is a multiple of 2 for SimpleGraph
+            if typeof(g) <: SimpleGraph
```
Comment (FHell) [inline src/NetworkStructures.jl:107]: Suggestion: Change this to
if is_directed(g) ....

**Comment by FHell** [comment]
Okay to merge from my side once documentation is updated and is_directed check is adjusted.

File: Project.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name = "NetworkDynamics"
 uuid = "22e9dc34-2a0d-11e9-0de0-8588d035468b"
 authors = ["Frank Hellmann <hellmann@pik-potsdam.de>, Michael Lindner <michaellindner@pik-potsdam.de>"]
-version = "0.4.2"
+version = "0.5.0-pre"
```
Comment (FHell) [inline Project.toml:4]: -> "0.5.0"

**Comment by lindnemi** [comment]
Merging this already now to avoid confusion with respect to the paper on arxiv. The remaining changes are more or less "cosmetic" and can go into v0.5.1

**Pull request merged**